README.md

```diff
@@ -1,57 +1,297 @@
+      ref.publisher = 'canonical'
+      ref.offer = 'UbuntuServer'
+      ref.sku = '16.04.0-LTS'
+      ref.version = 'latest'
+    end
+    store_profile.os_disk = ComputeModels::OSDisk.new.tap do |os_disk|
+      os_disk.name = "sample-os-disk-#{vm_name}"
+      os_disk.caching = ComputeModels::CachingTypes::None
+      os_disk.create_option = ComputeModels::DiskCreateOptionTypes::FromImage
+      os_disk.vhd = ComputeModels::VirtualHardDisk.new.tap do |vhd|
+        vhd.uri = "https://#{storage_acct.name}.blob.core.windows.net/rubycontainer/#{vm_name}.vhd"
+      end
+    end
+  end
+
+  vm.hardware_profile = ComputeModels::HardwareProfile.new.tap do |hardware|
+    hardware.vm_size = ComputeModels::VirtualMachineSizeTypes::StandardDS2V2
+  end
+
+  vm.network_profile = ComputeModels::NetworkProfile.new.tap do |net_profile|
+    net_profile.network_interfaces = [
+        ComputeModels::NetworkInterfaceReference.new.tap do |ref|
+          ref.id = nic.id
+          ref.primary = true
+        end
+    ]
+  end
+
+  # Use System Assigned Identity for the VM
+  vm.identity = ComputeModels::VirtualMachineIdentity.new.tap do |identity|
+    identity.type = ComputeModels::ResourceIdentityType::SystemAssigned
+  end
+end
+
+ssh_pub_location = File.expand_path('~/.ssh/id_rsa.pub')
+if File.exists? ssh_pub_location
+  puts "Found SSH public key in #{ssh_pub_location}. Disabling password and enabling SSH authentication."
+  key_data = File.read(ssh_pub_location)
+  puts "Using public key: #{key_data}"
+  vm_create_params.os_profile.linux_configuration = ComputeModels::LinuxConfiguration.new.tap do |linux|
+    linux.disable_password_authentication = true
+    linux.ssh = ComputeModels::SshConfiguration.new.tap do |ssh_config|
+      ssh_config.public_keys = [
+          ComputeModels::SshPublicKey.new.tap do |pub_key|
+            pub_key.key_data = key_data
+            pub_key.path = '/home/notAdmin/.ssh/authorized_keys'
+          end
+      ]
+    end
+  end
+end
+
+print_item vm = compute_client.virtual_machines.create_or_update(GROUP_NAME, "sample-ruby-vm-#{vm_name}", vm_create_params)
+```
+
+<a id="extension"></a>
+### Add MSI extension to the VM
+
+```ruby
+puts "Install Managed Service Identity Extension"
+ext_name = 'msiextension'
+vm_extension = ComputeModels::VirtualMachineExtension.new.tap do |extension|
+  extension.publisher = 'Microsoft.ManagedIdentity'
+  extension.virtual_machine_extension_type = 'ManagedIdentityExtensionForLinux'
+  extension.type_handler_version = '1.0'
+  extension.auto_upgrade_minor_version = true
+  extension.settings = Hash.new.tap do |settings|
+    settings['port'] = '50342'
+  end
+  extension.location = WEST_US
+end
+
+vm_ext = compute_client.virtual_machine_extensions.create_or_update(GROUP_NAME, "sample-ruby-vm-#{vm_name}", ext_name, vm_extension)
+```
+
+<a id="msi-extension"></a>
+### Verify MSI extension is running on VM by logging-in via ssh
+
```

Comment:
probably a good idea to indicate what to look for? and why? 
I see that in the actual code, we have a comment that says, run this command and look for X.